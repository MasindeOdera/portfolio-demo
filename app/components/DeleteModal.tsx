import styled from 'styled-components';
import { theme } from '../styles/theme';

interface DeleteModalProps {
  projectTitle: string;
  onCancel: () => void;
  onConfirm: () => void;
}

const ModalContainer = styled.div`
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  background-color: ${theme.colors.text}; //address the font white issue
  display: flex;
  flex-direction: column;
  width: 300px;
  height: 220px;
  padding: 20px;
  border-radius: 10px;
  box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
  z-index: 1000;
`;

export default function DeleteModal({ projectTitle, onCancel, onConfirm }: DeleteModalProps) {
  return (
    <ModalContainer>
      <p>Are you sure you want to delete &quot;{projectTitle}&quot;?</p>
      <button onClick={onConfirm}>Yes</button>
      <button onClick={onCancel}>No</button>
    </ModalContainer>
  );
}
