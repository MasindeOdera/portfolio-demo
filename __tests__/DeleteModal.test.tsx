import { render, screen, fireEvent } from '@testing-library/react';
import DeleteModal from '@/app/components/DeleteModal';

describe('DeleteModal Component', () => {
  const mockOnCancel = jest.fn();
  const mockOnConfirm = jest.fn();

  beforeEach(() => {
    render(
      <DeleteModal 
        projectTitle="Test Project" 
        onCancel={mockOnCancel} 
        onConfirm={mockOnConfirm} 
      />
    );
  });

  test('matches snapshot', () => {
    const { asFragment } = render(
      <DeleteModal 
        projectTitle="Test Project" 
        onCancel={mockOnCancel} 
        onConfirm={mockOnConfirm} 
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('displays the correct project title', () => {
    expect(screen.getByText(/Are you sure you want to delete "Test Project"?/i)).toBeInTheDocument();
  });

  test('calls onCancel when No button is clicked', () => {
    const noButton = screen.getByText(/No/i);
    fireEvent.click(noButton);
    expect(mockOnCancel).toHaveBeenCalled();
  });

  test('calls onConfirm when Yes button is clicked', () => {
    const yesButton = screen.getByText(/Yes/i);
    fireEvent.click(yesButton);
    expect(mockOnConfirm).toHaveBeenCalled();
  });
});
