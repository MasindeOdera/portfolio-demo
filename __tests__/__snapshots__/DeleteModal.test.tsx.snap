// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteModal Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-blHHSb bkqoCW"
  >
    <h2
      class="sc-gtLWhw jidGLU"
    >
      Delete Project
    </h2>
    <div
      class="sc-egkSDF leRVCI"
    >
      <p
        class="sc-fAUdSK lhKaBK"
      >
        Are you sure you want to delete "Test Project"?
      </p>
      <div
        class="sc-dntaoT eEcZmU"
      >
        <button
          class="sc-ivxoEo KmYNu"
        >
          Yes
        </button>
        <button
          class="sc-ivxoEo KmYNu"
        >
          No
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
